bin/jobsub_cmd

```diff
@@ -17,37 +17,79 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-schedd=""
-case "x$*" in
-*@*) schedd=`echo "$*" | sed -e 's/.*@//' -e 's/ .*//'`
-     args=`echo "$*" | sed -e 's/@[^ ]*//'`
-     ;;
-*)  args="$*"
-    ;;
-esac
-if [ x$schedd != x ]
-then
-    args="-name $schedd $args"
-fi
-
-# Check group and pull it into environment
-case "x$args" in
-*--group*)
-    group=`echo "$args" | sed -e 's/.*--group[=( )+]//' -e 's/ .*//'`
-    args=`echo "$args*" | sed -e "s/--group[=( )+]$group//"`
-    ;;
-*-G*)
-    group=`echo "$args" | sed -e 's/.*-G[( )+]//' -e 's/ .*//'`
-    args=`echo "$args" | sed -e "s/-G[( )+]$group//"`
-    ;;
-esac
-if [ x$group != x ]
-then
-    export GROUP=$group
-fi
-
-export X509_USER_PROXY=`fake_ifdh getProxy`
-export BEARER_TOKEN_FILE=`fake_ifdh getToken`
-
-cmd=`basename $0 | sed -e 's;^[a-z]*_;condor_;'`
-exec /usr/bin/$cmd -name $schedd $args
+import argparse
+import os
+import subprocess
+import sys
+
+PREFIX = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+sys.path.append(os.path.join(PREFIX, "lib"))
+
+import fake_ifdh
+
+class StoreGroupinEnvironment(argparse.Action):
+    """Action to store the given group in the GROUP environment variable"""
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        os.environ["GROUP"] = values
+        setattr(namespace, self.dest, values)
+
+
+def backquote(cmd):
+    sp = subprocess.Popen(cmd, shell=True, encoding='UTF-8', stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    res, stderr = sp.communicate()
+    return res
+
+def main():
+    parser = argparse.ArgumentParser(description = 'Condor command wrapper')
+    parser.add_argument(
+        "-G",
+        "--group",
+        help="Group/Experiment/Subgroup for priorities and accounting",
+        action=StoreGroupinEnvironment,
+    )
+    parser.add_argument(
+        "--role", help="VOMS Role for priorities and accounting", default="Analysis"
+    )
+
+
+    arglist,passthru = parser.parse_known_args()
+
+    # make list of arguments to pass to condor command:
+    # - the passthru arguments from above, except if we have
+    #   any 234@schedd style arguments, pick out the schedd and
+    #   keep the 234, and pass --name schedd as well
+    execargs = []
+    schedd = None
+    for i in passthru:
```

Comment:
What do you think about making this some sort of regex with capturing groups rather than finding "@" and splitting?  I'm worried about some rogue command where someone throws an "@" in there and it'll break this.  

That's just my personal preference, since it makes clear to future us exactly what's being parsed here, and what we're expecting.

Reply:
That might be prettier.  Then we can also make sure it's like [\d\.]+@[\w\.]+ or something (i.e. its digits and dots then a hostname...

Reply:
Done.